src/main/java/io/curity/identityserver/plugin/github/authentication/CallbackRequestHandler.java

```diff
@@ -194,40 +232,25 @@ private WebServiceClient getWebServiceClient(String uri)
         return data;
     }
 
-//    private void checkUserOrganizationMembership(Optional<AuthenticationResult> authenticationResult, String
-//            accessToken)
-//    {
-//        Attribute loginAttrubute = authenticationResult.get().getAttributes().getSubjectAttributes().get(LOGIN);
-//        String username = null;
-//        if (loginAttrubute != null)
-//        {
-//            username = loginAttrubute.getValue().toString();
-//        }
-//        if (notNullOrEmpty(username) && notNullOrEmpty(_config.getOrganizationName()))
-//        {
-//            HttpGet getRequest = new HttpGet(ORGANIZATION_MEMBER_CHECK_URL + _config.getOrganizationName() +
-//                    "/members/" + username);
-//            getRequest.addHeader(AUTHORIZATION, BEARER + accessToken);
-//
-//            try
-//            {
-//                HttpResponse response = _client.execute(getRequest);
-//                if (response.getStatusLine().getStatusCode() == HttpStatus.NOT_FOUND.getCode())
-//                {
-//                    _oauthClient.redirectToAuthenticationOnError("org_not_found", "ACCESS DENIED TO ORGANIZATION",
-//                            _config.id());
-//                }
-//
-//            }
-//            catch (IOException e)
-//            {
-//                _logger.warn("Could not communicate with organization endpoint", e);
-//
-//                throw _exceptionFactory.internalServerException(ErrorCode.EXTERNAL_SERVICE_ERROR, "Authentication " +
-//                        "failed");
-//            }
-//        }
-//    }
+    private void checkUserOrganizationMembership(String username, String accessToken)
+    {
+        _config.getManageOrganization().ifPresent(manageOrganization -> {
+            manageOrganization.getOrganizationName().ifPresent(organziationName -> {
+                HttpResponse tokenResponse = getWebServiceClient("https://api.github.com/orgs/" + organziationName + "/members/" + username)
+                        .request()
+                        .accept("application/json")
+                        .header("Authorization", "Bearer " + accessToken)
+                        .get()
+                        .response();
+                int statusCode = tokenResponse.statusCode();
+                if (tokenResponse.statusCode() == HttpStatus.NOT_FOUND.getCode()) {
+                    _logger.info("Got error response from user organization membership: error = {}", statusCode);
+
+                    throw _exceptionFactory.internalServerException(ErrorCode.EXTERNAL_SERVICE_ERROR);
```

Comment:
this should probably be a forbidden error and not a server error.

Reply:
okay, updated to `ACCESS_DENIED`